Rainbird

jbarrancos opened the pull request:
## Description:

Added rainsensor

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#4094

## Checklist:

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

jbarrancos committed:
Added rainsensor

jbarrancos committed:
Added rainsensor

jbarrancos committed:
Fixed issues

jbarrancos committed:
Fixed issues

jbarrancos committed:
Import wrong

jbarrancos committed:
Import wrong

jbarrancos committed:
Correct pyrainbird version

jbarrancos committed:
Correct pyrainbird version

jbarrancos committed:
Merge branch 'dev' into rainbird

bachya committed:
Updated codeowner for Tile device tracker (#10861)

MartinHjelmare commented on homeassistant/components/rainbird.py:
```diff
@@ -0,0 +1,49 @@
+    password = conf.get(CONF_PASSWORD)
+
+    from pyrainbird import RainbirdController
+    controller = RainbirdController(_LOGGER)
```
Why pass the logger instance? The library should use its own logger.

MartinHjelmare replied:
It's customary that each module should implement a logger like this:
```py
_LOGGER = logging.getLogger(__name__)
```
That way the origin of the log will be clear.

jbarrancos replied:
Fixed

MartinHjelmare commented on homeassistant/components/rainbird.py:
```diff
@@ -0,0 +1,49 @@
+    initialstatus = controller.currentIrrigation()
+    if initialstatus == -1:
+        _LOGGER.error("Error getting state. Possible configuration issues")
+        raise PlatformNotReady
```
This won't work here since this is not a platform being set up, but a component.

MartinHjelmare replied:
Just return `False` instead.

jbarrancos replied:
Fixed

MartinHjelmare commented on homeassistant/components/sensor/rainbird.py:
```diff
@@ -0,0 +1,81 @@
+from homeassistant.helpers.entity import Entity
+
+DEPENDENCIES = ['rainbird']
+DATA_RAINBIRD = 'rainbird'
```
Import this from the component.

jbarrancos replied:
fixed

MartinHjelmare commented on homeassistant/components/sensor/rainbird.py:
```diff
@@ -0,0 +1,81 @@
+                RainBirdSensor(controller, sensor_type))
+
+    add_devices(sensors, True)
+    return True
```
Platforms should not return anything.

jbarrancos replied:
Fixed

MartinHjelmare commented on homeassistant/components/sensor/rainbird.py:
```diff
@@ -0,0 +1,81 @@
+        """Initialize rain bird sensor."""
+        self._sensor_type = sensor_type
+        self._controller = controller
+        self._name = SENSOR_TYPES.get(self._sensor_type)[0]
```
```py
self._name = SENSOR_TYPES[self._sensor type][0]
```

jbarrancos replied:
Fixed

MartinHjelmare commented on homeassistant/components/sensor/rainbird.py:
```diff
@@ -0,0 +1,81 @@
+        self._sensor_type = sensor_type
+        self._controller = controller
+        self._name = SENSOR_TYPES.get(self._sensor_type)[0]
+        self._icon = 'mdi:{}'.format(SENSOR_TYPES.get(self._sensor_type)[2])
```
Same as above.

jbarrancos replied:
Fixed

MartinHjelmare commented on homeassistant/components/sensor/rainbird.py:
```diff
@@ -0,0 +1,81 @@
+    @property
+    def unit_of_measurement(self):
+        """Return the units of measurement."""
+        return SENSOR_TYPES.get(self._sensor_type)[1]
```
Same as above.

jbarrancos replied:
Fixed

MartinHjelmare commented on homeassistant/components/sensor/rainbird.py:
```diff
@@ -0,0 +1,81 @@
+Support for Rain Bird Irrigation system LNK WiFi Module.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/rainbird/
```
Update link.

jbarrancos replied:
The link will work with the new docs? As there will be default component docs now

jbarrancos replied:
Fixed

Julius2342 committed:
Revert "KNX: Added config option for broadcasting current time to KNX bus. (#10654)" (#10874)

This reverts commit cadd797200e339e165dbeff152169e4f4d716e2d.

As discussed within #10708 we should chose a different implementation. Therefore we should revert this change to avoid a breaking change.

fabaff committed:
Upgrade distro to 1.1.0 (#10850)

pvizeli committed:
Bugfix trigger state with multible entities (#10857)

* Bugfix trigger state with multible entities

* Fix numeric state

* fix lint

* fix dict

* fix unsub

* fix logic

* fix name

* fix new logic

* add test for state

* add numeric state test for unsub

* add test for multible entities

* Update numeric_state.py

* Update numeric_state.py

* Update state.py

* Fix logic for triple match

* Add clear to numeric state

* clear for state trigger

pvizeli committed:
tellstick fix DEPENDENCIES and update tellcore-net (#10859)

* Update requirements_all.txt

* Update tellstick.py

* Fix DEPENDENCIES

* Update requirements_all.txt

* fix format

* fix lint

* fix lint

* Update tellstick.py

* update tellcore-net

* update tellcore-net

* besser validate

balloob committed:
Update frontend to 20171130.0

fabaff committed:
Upgrade aiohttp to 2.3.5 (#10889)

fabaff committed:
Upgrade fastdotcom to 0.0.3 (#10886)

fabaff committed:
Upgrade schiene to 0.19 (#10887)

rytilahti committed:
Xiaomi Vacuum: remove deprecated calls (#10839)

* vacuum.xiaomi_miio: read dnd status properly instead of using imprecise dnd flag from vacuum_state

* vacuum.xiaomi_miio: use miio package instead of mirobo

* check only that wanted calls have taken place, ignore order of calls

* Fix linting issues

* Remove empty line after docstring

PhracturedBlue committed:
Create ecobee weather platform (#10869)

* Create ecobee weather component

* Update requirements_all for ecobee

* Fix missed lint issue

jeroenterheerdt committed:
Microsoft Text-to-speech: Fixing missing en-gb support bug (#10429)

* Fixing missing en-gb support bug

* Microsoft TTS adding support for rate, volume, pitch and contour.

* Microsoft TTS fixing support for jp-jp.

* Fixing linting error on line 67

* make impossible things possible :tada:

fabaff committed:
Upgrade youtube_dl to 2017.11.26 (#10890)

fabaff committed:
Upgrade yarl to 0.15.0 (#10888)

pvizeli committed:
Fix tests (#10891)

GenericStudent committed:
Refactored WHOIS sensor to resolve assumed key errors (#10662)

* Refactored WHOIS sensor to resolve assumed key errors

Altered it to now set an attribute key and value only if the attribute
is present in the WHOIS response. This prevents assumed keys (registrar)
from raising a KeyError on WHOIS lookups that don't contain registrar
information (onet.pl, wp.pl, for example).

* Removed non-used self._data

* WHOIS sensor now creates a new local attributes dict and overrides

* Corrected typos, refactored error cases to clear state adn attributes

* Resolved double return and refactored error logging

andrey-git committed:
Serve latest extra_html in dev mode (#10863)

andrey-git committed:
Reload groups after saving a change via config API (#10877)

fabaff committed:
Version bump to 0.59.0

PhracturedBlue committed:
Update ecobee version to fix stack-trace issue (#10894)

raymccarthy committed:
Pybotvac multi (#10843)

* Update requirements_all.txt

* Update neato.py

PhracturedBlue committed:
Fix issues from review of ecobee weather component (#10903)

* Fix issues from review

* Don't use STATE_UNKNOWN

PhracturedBlue committed:
Fix issues from review of ecobee weather component (#10903)

* Fix issues from review

* Don't use STATE_UNKNOWN

turbokongen committed:
Bugfix #10902 (#10904)

turbokongen committed:
Bugfix #10902 (#10904)

fabaff committed:
Merge pull request #10898 from home-assistant/release-0-59

0.59

fabaff committed:
Merge branch 'master' into dev

bonzini committed:
More declarative timeout syntax for manual alarm control panel. (#10738)

More declarative timeout syntax for manual alarm control panel

nickovs committed:
Unpacking RESTful sensor JSON results into attributes. (#10753)

* Added support for extracting JSON attributes from RESTful values

Setting the json_attributes configuration option to true on the
RESTful sensor will cause the result of the REST request to be parsed
as a JSON string and if successful the resulting dictionary will be
used for the attributes of the sensor.

* Added support for extracting JSON attributes from RESTful values

Setting the json_attributes configuration option to true on the
RESTful sensor will cause the result of the REST request to be parsed
as a JSON string and if successful the resulting dictionary will be
used for the attributes of the sensor.

* Added requirement that RESTful JSON results used as attributes must be
objects, not lists.

* Expanded test coverage to test REFTful JSON attributes with and
without a value template.

* Added support for extracting JSON attributes from RESTful values

Setting the json_attributes configuration option to true on the
RESTful sensor will cause the result of the REST request to be parsed
as a JSON string and if successful the resulting dictionary will be
used for the attributes of the sensor.

* Added requirement that RESTful JSON results used as attributes must be
objects, not lists.

* Expanded test coverage to test REFTful JSON attributes with and
without a value template.

* sensor.envirophat: add missing requirement (#7451)

Adding requirements that is not explicitly pulled in by the library
that manages the Enviro pHAT.

* PyPI Openzwave (#7415)

* Remove default zwave config path

PYOZW now has much more comprehensive default handling for the config
path (in src-lib/libopenzwave/libopenzwave.pyx:getConfig()). It looks in
the same place we were looking, plus _many_ more. It will certainly do a
much better job of finding the config files than we will (and will be
updated as the library is changed, so we don't end up chasing it). The
getConfig() method has been there for a while, but was subsntially
improved recently.

This change simply leaves the config_path as None if it is not
specified, which will trigger the default handling in PYOZW.

* Install python-openzwave from PyPI

As of version 0.4, python-openzwave supports installation from PyPI,
which means we can use our 'normal' dependency management tooling to
install it. Yay.

This uses the default 'embed' build (which goes and downloads
statically sources to avoid having to compile anything locally). Check
out the python-openzwave readme for more details.

* Add python-openzwave deps to .travis.yml

Python OpenZwave require the libudev headers to build. This adds the
libudev-dev package to Travis runs via the 'apt' addon for Travis.

Thanks to @MartinHjelmare for this fix.

* Update docker build for PyPI openzwave

Now that PYOZW can be install from PyPI, the docker image build process
can be simplified to remove the explicit compilation of PYOZW.

* Add datadog component (#7158)

* Add datadog component

* Improve test_invalid_config datadog test

* Use assert_setup_component for test setup

* Fix object type for default KNX port

#7429 describes a TypeError that is raised if the port is omitted in the config for the KNX component (integer is required (got type str)). This commit changes the default port from a string to an integer. I expect this will resolve that issue...

* Added support for extracting JSON attributes from RESTful values

Setting the json_attributes configuration option to true on the
RESTful sensor will cause the result of the REST request to be parsed
as a JSON string and if successful the resulting dictionary will be
used for the attributes of the sensor.

* Added requirement that RESTful JSON results used as attributes must be
objects, not lists.

* Expanded test coverage to test REFTful JSON attributes with and
without a value template.

* Added support for extracting JSON attributes from RESTful values

Setting the json_attributes configuration option to true on the
RESTful sensor will cause the result of the REST request to be parsed
as a JSON string and if successful the resulting dictionary will be
used for the attributes of the sensor.

* Added requirement that RESTful JSON results used as attributes must be
objects, not lists.

* Expanded test coverage to test REFTful JSON attributes with and
without a value template.

* Added support for extracting JSON attributes from RESTful values

Setting the json_attributes configuration option to true on the
RESTful sensor will cause the result of the REST request to be parsed
as a JSON string and if successful the resulting dictionary will be
used for the attributes of the sensor.

* Added requirement that RESTful JSON results used as attributes must be
objects, not lists.

* Expanded test coverage to test REFTful JSON attributes with and
without a value template.

* Fixed breaks cause by manual upstream merge.

* Added one extra blank line to make PyLint happy.

* Switched json_attributes to be a list of keys rather than a boolean.

The value of json_attributes can now be either a comma sepaated list
of key names or a YAML list of key names. Only matching keys in a
retuned JSON dictionary will be mapped to sensor attributes.

Updated test cases to handle json_attributes being a list.

Also fixed two minor issues arrising from manual merge with 0.58 master.

* Added an explicit default value to the json_attributes config entry.

* Removed self.update() from __init__() body.

* Expended unit tests for error cases of json_attributes processing.

* Align quotes

fabaff committed:
Bump dev to 0.60.0.dev0 (#10912)

molobrakos committed:
Update eliqonline.py (#10914)

Channel id is now required (change in API)

RyuzakiKK committed:
Add iAlarm support (#10878)

* Add iAlarm support

* Minor fixes to iAlarm

* Rename ialarmpanel to ialarm and add a check for the host value

* corrections in the value validation of ialarm

* add a missing period on ialarm

RiRomain committed:
Correction of Samsung Power OFF behaviour (#10907)

* Correction of Samsung Power OFF behaviour

Addition of a delay after powering OFF a Samsung TV, this avoid status
update from powering the TV back ON.
Deletion of update() return statement, return value not used.

* Rename self._end_of_power_off_command into self._end_of_power_off

* Removal of unused line break in Samsung TV component

fabaff committed:
Add Alpha Vantage sensor (#10873)

* Add Alpha Vantage sensor

* Remove data object

* Remove unused vars and change return

* Fix typo

Commit:
Don't repeat getting receiver name on each update / pushed to denonavr 0.5.5 (#10915)

Commit:
Add Min and Event Count Metrics To Prometheus (#10530)

* Added min and Events sensor types to prometheus

* Updated prometheus client and fixed invalid swith state

* Added metric to count number of times an automation is triggered

* Removed assumption that may not apply to everybody

* Fixed tests

* Updated requirements_test_all

* Fixed unit tests

wrboyce committed:
fix ios component config generation (#10923)

Fixes: #19020

wrboyce committed:
fix ios component config generation (#10923)

Fixes: #19020

danielperna84 committed:
Fix Notifications for Android TV (#10798)

* Fixed icon path, added dynamic icon

* Addressing requested changes

* Using DEFAULT_ICON

* Using CONF_ICON from const

* Getting hass_frontend path via import

* Lint

* Using embedded 1px transparent icon

* woof -.-

* Lint

danielperna84 committed:
Fix Notifications for Android TV (#10798)

* Fixed icon path, added dynamic icon

* Addressing requested changes

* Using DEFAULT_ICON

* Using CONF_ICON from const

* Getting hass_frontend path via import

* Lint

* Using embedded 1px transparent icon

* woof -.-

* Lint

balloob committed:
Update coveragerc (#10931)

* Sort coveragerc

* Add climate.honeywell and vacuum.xiaomi_miio to coveragerc

balloob committed:
Update frontend to 20171204.0 (#10934)

balloob committed:
Update frontend to 20171204.0 (#10934)

craigjmidwinter committed:
Dominos no order fix (#10935)

* check for none

* fix error from no store being set

* typo

* Lint

* fix default as per notes. Lint fix and make closest store None not False

* update default

craigjmidwinter committed:
Dominos no order fix (#10935)

* check for none

* fix error from no store being set

* typo

* Lint

* fix default as per notes. Lint fix and make closest store None not False

* update default

balloob committed:
Version bump to 0.59.1

balloob committed:
Merge pull request #10936 from home-assistant/release-0-59-1

0.59.1

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

DanNixon committed:
Report availability of TP-Link smart sockets (#10933)

* Report availability of TP-Link smart sockets

* Changes according to our style guide

nbougues committed:
Set percent unit for battery level so that history displays properly; edited variable name for consistency (#10932)

hudashot committed:
Export climate status and target temperature to Prometheus (#10919)

* Export climate metrics to Prometheus.

This adds climate_state and temperature_c metrics for each climate
device.

* Add more climate states to state_as_number

dasos committed:
Tado ignore invalid devices (#10927)

* Ignore devices without temperatures

* Typo

* Linting

* Removing return false

* Another typo. :(

* Spelling received correctly

molobrakos committed:
Upgrade tellduslive library version (closes https://github.com/home-assistant/home-assistant/issues/10922) (#10950)

molobrakos committed:
Upgrade tellduslive library version (closes https://github.com/home-assistant/home-assistant/issues/10922) (#10950)

PaulAnnekov committed:
don't ignore voltage data if sensor data changed (#10925)

mateuszdrab committed:
Fix linksys_ap.py by inheriting DeviceScanner (#10947)

As per issue #8638, the class wasn't inheriting from DeviceScanner, this commit patches it up.

mateuszdrab committed:
Fix linksys_ap.py by inheriting DeviceScanner (#10947)

As per issue #8638, the class wasn't inheriting from DeviceScanner, this commit patches it up.

Commit:
Add ADS component (#10142)

* add ads hub, light and switch

* add binary sensor prototype

* switch: use adsvar for connection

* fix some issues with binary sensor

* fix binary sensor

* fix all platforms

* use latest pyads

* fixed error with multiple binary sensors

* add sensor

* add ads sensor

* clean up after shutdown

* ads component with platforms switch, binary_sensor, light, sensor

add locking

poll sensors at startup

update state of ads switch and light

update ads requirements

remove update() from constructors on ads platforms

omit ads coverage

ads catch read error when polling

* add ads service

* add default settings for use_notify and poll_interval

* fix too long line

* Fix style issues

* no pydocstyle errors

* Send and receive native brightness data to ADS device to prevent issues with math.floor reducing brightness -1 at every switch

* Enable non dimmable lights

* remove setting of self._state in switch

* remove polling

* Revert "remove polling"

This reverts commit 7da420f82385a4a5c66a929af7025c00ed197e86.

* add service schema, add links to documentation

* fix naming, cleanup

* re-remove polling

* use async_added_to_hass for setup of callbacks

* fix comment.

* add callbacks for changed values

* use async_add_job for creating device notifications

* set should_poll to False for all platforms

* change should_poll to property

* add service description to services.yaml

* add for brigthness not being None

* put ads component in package

* Remove whitespace

* omit ads package

craigjmidwinter committed:
Reload closest store on api menu request (#10962)

* reload closest store on api request

* revert change from debugging

craigjmidwinter committed:
Reload closest store on api menu request (#10962)

* reload closest store on api request

* revert change from debugging

HerrHofrat committed:
Gearbest sensor (#10556)

* Added Gearbest Sensor

* Updated required files

* Fixed houndci-bout findings

* Fix tox lint errors

* Changed code according to review
Implemented library version 1.0.5

* Fixed houndci-bot findings

* Fixed tox lint issues

* Updated item schema to use has_at_least_one_key
Added conf constants

* Remove CONF_ constants and import them from homeassistant.const

* Removed CurrencyConverter from hass
Fixed couple of issues found by MartinHjelmare

b10m committed:
Add Ziggo Mediabox XL media_player (#10514)

* Add Ziggo Mediabox XL media_player

* Using pypi module ziggo-mediabox-xl now.

* Code review changes

ziotibia81 committed:
Generic thermostat initial_operation_mode (#10690)

* Generic thermostat restore operation mode

* Test restore operation mode

* Fix trailing whitespace

* Fix line too long

* Fix test duplicate entity_id

* Fix test

* async_added_to_hass modify modify internal state

* Test inital_operation_mode

* More restore state tests

* Fix whitespace

* fix test_custom_setup_param

* Test "None" target temp

emlove committed:
Use new build path for dev translations (#10937)

masarliev committed:
Add option to set default hide if away for new devices (#10762)

* Option to change hide_if_away

* tests fix

* change new device defaults

* tests and requested changes

* fix assert

Commit:
Allow chime to work for wink siren/chime (#10961)

* Allow Wink siren/chimes to work

* Updated requirements_all.txt

Commit:
Allow chime to work for wink siren/chime (#10961)

* Allow Wink siren/chimes to work

* Updated requirements_all.txt

balloob committed:
Revert pychromecast update (#10989)

* Revert pychromecast update

* Update cast.py

balloob committed:
Revert pychromecast update (#10989)

* Revert pychromecast update

* Update cast.py

andrey-git committed:
Require FF43 for latest js (#10941)

* Require FF43 for latest js

`Array.prototype.includes` added in Firefox 43

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

* Update __init__.py

andrey-git committed:
Require FF43 for latest js (#10941)

* Require FF43 for latest js

`Array.prototype.includes` added in Firefox 43

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

* Update __init__.py

balloob committed:
Version bump to 0.59.2

balloob committed:
Merge pull request #10990 from home-assistant/release-0-59-2

0.59.2

balloob committed:
Merge branch 'master' into dev

DanNixon committed:
Allow disabling the LEDs on TP-Link smart plugs (#10980)

balloob committed:
Update frontend to 20171206.0

masarliev committed:
Meraki AP Device tracker (#10971)

* Device tracker for meraki AP

* styles fix

* fix again

* again

* and again :)

* fix hide if away

* docs and optimization

* tests and fixes

* styles

* styles

* styles

* styles

* styles fix. Hope last

* clear track new

* changes

* fix accuracy error and requested changes

* remove meraki from .coveragerc

* tests and minor changes

* remove location

pvizeli committed:
Update tradfri.py (#10991)

OddBloke committed:
webostv: Ensure source exists before use (#10959)

In a case where either (a) an incorrect source name is used, or (b) the
TV isn't currently queryable (e.g. it's off), we get tracebacks because
we assume the source that we are being asked to select exists in
self._source_list.

This makes the lookup code a little more rugged, and adds in a warning
message (in place of the current exception).

Commit:
Ensure Docker script files uses LF line endings to support Docker for Windows. (#10067)

alanfischer committed:
Added Vera scenes (#10424)

* Added Vera scenes

* Fixed flake8 issues

* Fixed comments

* Moved vera to use hass.data

* Made requested changes

jeroenterheerdt committed:
Fix Egardia alarm status shown as unknown after restart (#11010)

jbarrancos committed:
Fixed issues from reviews

jbarrancos committed:
Fixed issues from reviews

jbarrancos committed:
Merge branch 'rainbird' of https://github.com/jbarrancos/home-assistant into rainbird

jbarrancos committed:
Syncup (#9)

* Updated codeowner for Tile device tracker (#10861)

* Revert "KNX: Added config option for broadcasting current time to KNX bus. (#10654)" (#10874)

This reverts commit cadd797200e339e165dbeff152169e4f4d716e2d.

As discussed within #10708 we should chose a different implementation. Therefore we should revert this change to avoid a breaking change.

* Upgrade distro to 1.1.0 (#10850)

* Bugfix trigger state with multible entities (#10857)

* Bugfix trigger state with multible entities

* Fix numeric state

* fix lint

* fix dict

* fix unsub

* fix logic

* fix name

* fix new logic

* add test for state

* add numeric state test for unsub

* add test for multible entities

* Update numeric_state.py

* Update numeric_state.py

* Update state.py

* Fix logic for triple match

* Add clear to numeric state

* clear for state trigger

* tellstick fix DEPENDENCIES and update tellcore-net (#10859)

* Update requirements_all.txt

* Update tellstick.py

* Fix DEPENDENCIES

* Update requirements_all.txt

* fix format

* fix lint

* fix lint

* Update tellstick.py

* update tellcore-net

* update tellcore-net

* besser validate

* Update frontend to 20171130.0

* Upgrade aiohttp to 2.3.5 (#10889)

* Upgrade fastdotcom to 0.0.3 (#10886)

* Upgrade schiene to 0.19 (#10887)

* Xiaomi Vacuum: remove deprecated calls (#10839)

* vacuum.xiaomi_miio: read dnd status properly instead of using imprecise dnd flag from vacuum_state

* vacuum.xiaomi_miio: use miio package instead of mirobo

* check only that wanted calls have taken place, ignore order of calls

* Fix linting issues

* Remove empty line after docstring

* Create ecobee weather platform (#10869)

* Create ecobee weather component

* Update requirements_all for ecobee

* Fix missed lint issue

* Microsoft Text-to-speech: Fixing missing en-gb support bug (#10429)

* Fixing missing en-gb support bug

* Microsoft TTS adding support for rate, volume, pitch and contour.

* Microsoft TTS fixing support for jp-jp.

* Fixing linting error on line 67

* make impossible things possible :tada:

* Upgrade youtube_dl to 2017.11.26 (#10890)

* Upgrade yarl to 0.15.0 (#10888)

* Fix tests (#10891)

* Refactored WHOIS sensor to resolve assumed key errors (#10662)

* Refactored WHOIS sensor to resolve assumed key errors

Altered it to now set an attribute key and value only if the attribute
is present in the WHOIS response. This prevents assumed keys (registrar)
from raising a KeyError on WHOIS lookups that don't contain registrar
information (onet.pl, wp.pl, for example).

* Removed non-used self._data

* WHOIS sensor now creates a new local attributes dict and overrides

* Corrected typos, refactored error cases to clear state adn attributes

* Resolved double return and refactored error logging

* Serve latest extra_html in dev mode (#10863)

* Reload groups after saving a change via config API (#10877)

* Version bump to 0.59.0

* Update ecobee version to fix stack-trace issue (#10894)

* Pybotvac multi (#10843)

* Update requirements_all.txt

* Update neato.py

* Fix issues from review of ecobee weather component (#10903)

* Fix issues from review

* Don't use STATE_UNKNOWN

* Bugfix #10902 (#10904)

* Fix issues from review of ecobee weather component (#10903)

* Fix issues from review

* Don't use STATE_UNKNOWN

* Bugfix #10902 (#10904)

* More declarative timeout syntax for manual alarm control panel. (#10738)

More declarative timeout syntax for manual alarm control panel

* Unpacking RESTful sensor JSON results into attributes. (#10753)

* Added support for extracting JSON attributes from RESTful values

Setting the json_attributes configuration option to true on the
RESTful sensor will cause the result of the REST request to be parsed
as a JSON string and if successful the resulting dictionary will be
used for the attributes of the sensor.

* Added support for extracting JSON attributes from RESTful values

Setting the json_attributes configuration option to true on the
RESTful sensor will cause the result of the REST request to be parsed
as a JSON string and if successful the resulting dictionary will be
used for the attributes of the sensor.

* Added requirement that RESTful JSON results used as attributes must be
objects, not lists.

* Expanded test coverage to test REFTful JSON attributes with and
without a value template.

* Added support for extracting JSON attributes from RESTful values

Setting the json_attributes configuration option to true on the
RESTful sensor will cause the result of the REST request to be parsed
as a JSON string and if successful the resulting dictionary will be
used for the attributes of the sensor.

* Added requirement that RESTful JSON results used as attributes must be
objects, not lists.

* Expanded test coverage to test REFTful JSON attributes with and
without a value template.

* sensor.envirophat: add missing requirement (#7451)

Adding requirements that is not explicitly pulled in by the library
that manages the Enviro pHAT.

* PyPI Openzwave (#7415)

* Remove default zwave config path

PYOZW now has much more comprehensive default handling for the config
path (in src-lib/libopenzwave/libopenzwave.pyx:getConfig()). It looks in
the same place we were looking, plus _many_ more. It will certainly do a
much better job of finding the config files than we will (and will be
updated as the library is changed, so we don't end up chasing it). The
getConfig() method has been there for a while, but was subsntially
improved recently.

This change simply leaves the config_path as None if it is not
specified, which will trigger the default handling in PYOZW.

* Install python-openzwave from PyPI

As of version 0.4, python-openzwave supports installation from PyPI,
which means we can use our 'normal' dependency management tooling to
install it. Yay.

This uses the default 'embed' build (which goes and downloads
statically sources to avoid having to compile anything locally). Check
out the python-openzwave readme for more details.

* Add python-openzwave deps to .travis.yml

Python OpenZwave require the libudev headers to build. This adds the
libudev-dev package to Travis runs via the 'apt' addon for Travis.

Thanks to @MartinHjelmare for this fix.

* Update docker build for PyPI openzwave

Now that PYOZW can be install from PyPI, the docker image build process
can be simplified to remove the explicit compilation of PYOZW.

* Add datadog component (#7158)

* Add datadog component

* Improve test_invalid_config datadog test

* Use assert_setup_component for test setup

* Fix object type for default KNX port

#7429 describes a TypeError that is raised if the port is omitted in the config for the KNX component (integer is required (got type str)). This commit changes the default port from a string to an integer. I expect this will resolve that issue...

* Added support for extracting JSON attributes from RESTful values

Setting the json_attributes configuration option to true on the
RESTful sensor will cause the result of the REST request to be parsed
as a JSON string and if successful the resulting dictionary will be
used for the attributes of the sensor.

* Added requirement that RESTful JSON results used as attributes must be
objects, not lists.

* Expanded test coverage to test REFTful JSON attributes with and
without a value template.

* Added support for extracting JSON attributes from RESTful values

Setting the json_attributes configuration option to true on the
RESTful sensor will cause the result of the REST request to be parsed
as a JSON string and if successful the resulting dictionary will be
used for the attributes of the sensor.

* Added requirement that RESTful JSON results used as attributes must be
objects, not lists.

* Expanded test coverage to test REFTful JSON attributes with and
without a value template.

* Added support for extracting JSON attributes from RESTful values

Setting the json_attributes configuration option to true on the
RESTful sensor will cause the result of the REST request to be parsed
as a JSON string and if successful the resulting dictionary will be
used for the attributes of the sensor.

* Added requirement that RESTful JSON results used as attributes must be
objects, not lists.

* Expanded test coverage to test REFTful JSON attributes with and
without a value template.

* Fixed breaks cause by manual upstream merge.

* Added one extra blank line to make PyLint happy.

* Switched json_attributes to be a list of keys rather than a boolean.

The value of json_attributes can now be either a comma sepaated list
of key names or a YAML list of key names. Only matching keys in a
retuned JSON dictionary will be mapped to sensor attributes.

Updated test cases to handle json_attributes being a list.

Also fixed two minor issues arrising from manual merge with 0.58 master.

* Added an explicit default value to the json_attributes config entry.

* Removed self.update() from __init__() body.

* Expended unit tests for error cases of json_attributes processing.

* Align quotes

* Bump dev to 0.60.0.dev0 (#10912)

* Update eliqonline.py (#10914)

Channel id is now required (change in API)

* Add iAlarm support (#10878)

* Add iAlarm support

* Minor fixes to iAlarm

* Rename ialarmpanel to ialarm and add a check for the host value

* corrections in the value validation of ialarm

* add a missing period on ialarm

* Correction of Samsung Power OFF behaviour (#10907)

* Correction of Samsung Power OFF behaviour

Addition of a delay after powering OFF a Samsung TV, this avoid status
update from powering the TV back ON.
Deletion of update() return statement, return value not used.

* Rename self._end_of_power_off_command into self._end_of_power_off

* Removal of unused line break in Samsung TV component

* Add Alpha Vantage sensor (#10873)

* Add Alpha Vantage sensor

* Remove data object

* Remove unused vars and change return

* Fix typo

* Don't repeat getting receiver name on each update / pushed to denonavr 0.5.5 (#10915)

* Add Min and Event Count Metrics To Prometheus (#10530)

* Added min and Events sensor types to prometheus

* Updated prometheus client and fixed invalid swith state

* Added metric to count number of times an automation is triggered

* Removed assumption that may not apply to everybody

* Fixed tests

* Updated requirements_test_all

* Fixed unit tests

* fix ios component config generation (#10923)

Fixes: #19020

* Fix Notifications for Android TV (#10798)

* Fixed icon path, added dynamic icon

* Addressing requested changes

* Using DEFAULT_ICON

* Using CONF_ICON from const

* Getting hass_frontend path via import

* Lint

* Using embedded 1px transparent icon

* woof -.-

* Lint

* Update coveragerc (#10931)

* Sort coveragerc

* Add climate.honeywell and vacuum.xiaomi_miio to coveragerc

* Update frontend to 20171204.0 (#10934)

* Dominos no order fix (#10935)

* check for none

* fix error from no store being set

* typo

* Lint

* fix default as per notes. Lint fix and make closest store None not False

* update default

* Version bump to 0.59.1

* Fix Notifications for Android TV (#10798)

* Fixed icon path, added dynamic icon

* Addressing requested changes

* Using DEFAULT_ICON

* Using CONF_ICON from const

* Getting hass_frontend path via import

* Lint

* Using embedded 1px transparent icon

* woof -.-

* Lint

* fix ios component config generation (#10923)

Fixes: #19020

* Update frontend to 20171204.0 (#10934)

* Dominos no order fix (#10935)

* check for none

* fix error from no store being set

* typo

* Lint

* fix default as per notes. Lint fix and make closest store None not False

* update default

* Report availability of TP-Link smart sockets (#10933)

* Report availability of TP-Link smart sockets

* Changes according to our style guide

* Set percent unit for battery level so that history displays properly; edited variable name for consistency (#10932)

* Export climate status and target temperature to Prometheus (#10919)

* Export climate metrics to Prometheus.

This adds climate_state and temperature_c metrics for each climate
device.

* Add more climate states to state_as_number

* Tado ignore invalid devices (#10927)

* Ignore devices without temperatures

* Typo

* Linting

* Removing return false

* Another typo. :(

* Spelling received correctly

* Upgrade tellduslive library version (closes https://github.com/home-assistant/home-assistant/issues/10922) (#10950)

* don't ignore voltage data if sensor data changed (#10925)

* Fix linksys_ap.py by inheriting DeviceScanner (#10947)

As per issue #8638, the class wasn't inheriting from DeviceScanner, this commit patches it up.

* Add ADS component (#10142)

* add ads hub, light and switch

* add binary sensor prototype

* switch: use adsvar for connection

* fix some issues with binary sensor

* fix binary sensor

* fix all platforms

* use latest pyads

* fixed error with multiple binary sensors

* add sensor

* add ads sensor

* clean up after shutdown

* ads component with platforms switch, binary_sensor, light, sensor

add locking

poll sensors at startup

update state of ads switch and light

update ads requirements

remove update() from constructors on ads platforms

omit ads coverage

ads catch read error when polling

* add ads service

* add default settings for use_notify and poll_interval

* fix too long line

* Fix style issues

* no pydocstyle errors

* Send and receive native brightness data to ADS device to prevent issues with math.floor reducing brightness -1 at every switch

* Enable non dimmable lights

* remove setting of self._state in switch

* remove polling

* Revert "remove polling"

This reverts commit 7da420f82385a4a5c66a929af7025c00ed197e86.

* add service schema, add links to documentation

* fix naming, cleanup

* re-remove polling

* use async_added_to_hass for setup of callbacks

* fix comment.

* add callbacks for changed values

* use async_add_job for creating device notifications

* set should_poll to False for all platforms

* change should_poll to property

* add service description to services.yaml

* add for brigthness not being None

* put ads component in package

* Remove whitespace

* omit ads package

* Reload closest store on api menu request (#10962)

* reload closest store on api request

* revert change from debugging

* Gearbest sensor (#10556)

* Added Gearbest Sensor

* Updated required files

* Fixed houndci-bout findings

* Fix tox lint errors

* Changed code according to review
Implemented library version 1.0.5

* Fixed houndci-bot findings

* Fixed tox lint issues

* Updated item schema to use has_at_least_one_key
Added conf constants

* Remove CONF_ constants and import them from homeassistant.const

* Removed CurrencyConverter from hass
Fixed couple of issues found by MartinHjelmare

* Add Ziggo Mediabox XL media_player (#10514)

* Add Ziggo Mediabox XL media_player

* Using pypi module ziggo-mediabox-xl now.

* Code review changes

* Generic thermostat initial_operation_mode (#10690)

* Generic thermostat restore operation mode

* Test restore operation mode

* Fix trailing whitespace

* Fix line too long

* Fix test duplicate entity_id

* Fix test

* async_added_to_hass modify modify internal state

* Test inital_operation_mode

* More restore state tests

* Fix whitespace

* fix test_custom_setup_param

* Test "None" target temp

* Use new build path for dev translations (#10937)

* Add option to set default hide if away for new devices (#10762)

* Option to change hide_if_away

* tests fix

* change new device defaults

* tests and requested changes

* fix assert

* Allow chime to work for wink siren/chime (#10961)

* Allow Wink siren/chimes to work

* Updated requirements_all.txt

* Revert pychromecast update (#10989)

* Revert pychromecast update

* Update cast.py

* Require FF43 for latest js (#10941)

* Require FF43 for latest js

`Array.prototype.includes` added in Firefox 43

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

* Update __init__.py

* Version bump to 0.59.2

* Require FF43 for latest js (#10941)

* Require FF43 for latest js

`Array.prototype.includes` added in Firefox 43

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

* Update __init__.py

* Fix linksys_ap.py by inheriting DeviceScanner (#10947)

As per issue #8638, the class wasn't inheriting from DeviceScanner, this commit patches it up.

* Upgrade tellduslive library version (closes https://github.com/home-assistant/home-assistant/issues/10922) (#10950)

* Reload closest store on api menu request (#10962)

* reload closest store on api request

* revert change from debugging

* Allow chime to work for wink siren/chime (#10961)

* Allow Wink siren/chimes to work

* Updated requirements_all.txt

* Revert pychromecast update (#10989)

* Revert pychromecast update

* Update cast.py

* Allow disabling the LEDs on TP-Link smart plugs (#10980)

* Update frontend to 20171206.0

* Meraki AP Device tracker (#10971)

* Device tracker for meraki AP

* styles fix

* fix again

* again

* and again :)

* fix hide if away

* docs and optimization

* tests and fixes

* styles

* styles

* styles

* styles

* styles fix. Hope last

* clear track new

* changes

* fix accuracy error and requested changes

* remove meraki from .coveragerc

* tests and minor changes

* remove location

* Update tradfri.py (#10991)

* webostv: Ensure source exists before use (#10959)

In a case where either (a) an incorrect source name is used, or (b) the
TV isn't currently queryable (e.g. it's off), we get tracebacks because
we assume the source that we are being asked to select exists in
self._source_list.

This makes the lookup code a little more rugged, and adds in a warning
message (in place of the current exception).

* Ensure Docker script files uses LF line endings to support Docker for Windows. (#10067)

* Added Vera scenes (#10424)

* Added Vera scenes

* Fixed flake8 issues

* Fixed comments

* Moved vera to use hass.data

* Made requested changes

* Fix Egardia alarm status shown as unknown after restart (#11010)

jbarrancos committed:
Revert "Syncup (#9)" (#10)

This reverts commit f84ae3dfa9de6bf22d9948748b914a9d048a5ff1.

jbarrancos committed:
Revert "Syncup (#9)" (#10)

This reverts commit f84ae3dfa9de6bf22d9948748b914a9d048a5ff1.

jbarrancos committed:
Updated Requirements

jbarrancos committed:
Updated Requirements

jbarrancos committed:
Merge branch 'rainbird' of https://github.com/jbarrancos/home-assistant into rainbird

lwis committed:
Handle OSError when forcibly turning off media_player.samsungtv (#10997)

MartinHjelmare commented:
I think you should do an interactive rebase and completely remove the "Syncup" commits, both the first and the revert.

When you want to catch up on changes made on upstream dev, it's better to rebase your branch onto dev than merging dev into your branch:
https://home-assistant.io/developers/development_catching_up/

mar-schmidt committed:
Shuffle support in Sonos (#10875)

* initial commit of shuffle option for sonos

* added test

* Small adjustments to adhere to review requests

* Removed unnessesary setting of variable. Use shuffle state from soco instead

jbarrancos committed:
Merge branch 'rainbird' of https://github.com/jbarrancos/home-assistant into rainbird

jbarrancos commented:
I'll close this one and open a new one.. Once git is cleaned up on my side.. Sorry
